### PR TITLE
Deneb spectests release v1.4.0-beta.2-hotfix

### PR DESCRIPTION
**Pull request by potuz**
implement deneb DA forkchoice spectests. 

exports ReceiveBlob from the blockchain package

### CODE REVIEW

**Comment by terencechain** [comment]
This PR also consisted the following changes

- [x] https://github.com/ethereum/consensus-specs/pull/3352
- [x] https://eips.ethereum.org/EIPS/eip-7514
- [x] https://github.com/ethereum/consensus-specs/pull/3484

File: beacon-chain/core/epoch/epoch_processing.go [hunk 1]
```diff
@@ -142,6 +142,13 @@ func ProcessRegistryUpdates(ctx context.Context, state state.BeaconState) (state
 		return nil, errors.Wrap(err, "could not get churn limit")
 	}
 
+	if state.Version() >= version.Deneb {
+		// Cap churn limit to max per epoch churn limit. New in EIP7514.
+		if churnLimit > params.BeaconConfig().MaxPerEpochActivationChurnLimit {
+			churnLimit = params.BeaconConfig().MaxPerEpochActivationChurnLimit
+		}
+	}
+
```
Comment (prestonvanloon) [inline beacon-chain/core/epoch/epoch_processing.go:151]: Can we add unit tests for this? It's likely covered by spectest but we should have our own tests as well as spectests are a bit of a blackbox sometimes.

File: testing/spectest/minimal/deneb/forkchoice/forkchoice_test.go [hunk 2]
```diff
@@ -2,11 +2,9 @@ package forkchoice
...
 	"testing"
-
-	"github.com/prysmaticlabs/prysm/v4/runtime/version"
-	"github.com/prysmaticlabs/prysm/v4/testing/spectest/shared/common/forkchoice"
 )
 
 func TestMinimal_Deneb_Forkchoice(t *testing.T) {
-	forkchoice.Run(t, "minimal", version.Deneb)
+	// forkchoice.Run(t, "minimal", version.Deneb)
+	t.Skip("blocked by go-kzg-4844 minimal trusted setup")
```
Comment (terencechain) [inline testing/spectest/minimal/deneb/forkchoice/forkchoice_test.go:9]:
```suggestion
	t.Skip("blocked by go-kzg-4844 minimal trusted setup")
	forkchoice.Run(t, "minimal", version.Deneb)
```

**Comment by prestonvanloon** [comment]
Fails to build

```
ERROR: /buildkite/builds/buildkite-new-3-1/prysmatic-labs/prysm/config/params/BUILD.bazel:42:8: //config/params:go_default_test depends on @consensus_spec//:spec_data in repository @consensus_spec which failed to fetch. no such package '@consensus_spec//': java.io.IOException: Error extracting /root/.cache/bazel/_bazel_root/29b315a8c41f2348dc9cc735eb2d30a0/external/consensus_spec/temp7051674763010975498/v1.4.0-beta.2-hotfix.tar.gz to /root/.cache/bazel/_bazel_root/29b315a8c41f2348dc9cc735eb2d30a0/external/consensus_spec/temp7051674763010975498: Prefix "consensus-specs-1.4.0-beta.2-hotfix" was given, but not found in the archive. Here are possible prefixes for this archive: "consensus-specs-1.4.0-beta.2".
```

**Review by prestonvanloon — APPROVED** [review]

**Pull request merged**